sharding/expandedListSampling_test.go

```diff
@@ -0,0 +1,131 @@
+package sharding
+
+import (
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/hashing/sha256"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewSelectorExpandedListNilValidatorsShouldErr(t *testing.T) {
+	weights := createDummyWeights(10, 32, 2, nextWeightGeometricProgression)
+	hasher := &sha256.Sha256{}
+
+	selector, err := NewSelectorExpandedList(nil, weights, hasher)
+	require.Equal(t, ErrNilValidators, err)
+	require.Nil(t, selector)
+}
+
+func TestNewSelectorExpandedListNilWeightsShouldErr(t *testing.T) {
+	validators := createDummyNodesList(10, "shard_0")
+	hasher := &sha256.Sha256{}
+
+	selector, err := NewSelectorExpandedList(validators, nil, hasher)
+	require.Equal(t, ErrNilWeights, err)
+	require.Nil(t, selector)
+}
+
+func TestNewSelectorExpandedListNilHasherShouldErr(t *testing.T) {
+	validators := createDummyNodesList(10, "shard_0")
+	weights := createDummyWeights(10, 32, 2, nextWeightGeometricProgression)
+
+	selector, err := NewSelectorExpandedList(validators, weights, nil)
+	require.Equal(t, ErrNilHasher, err)
+	require.Nil(t, selector)
+}
+
+func TestNewSelectorExpandedListInvalidWeightShouldErr(t *testing.T) {
+	validators := createDummyNodesList(10, "shard_0")
+	weights := createDummyWeights(10, 32, 2, nextWeightGeometricProgression)
+	weights[1] = 0
+	hasher := &sha256.Sha256{}
+
+	selector, err := NewSelectorExpandedList(validators, weights, hasher)
+	require.Equal(t, ErrInvalidWeight, err)
+	require.Nil(t, selector)
+}
+
+func TestNewSelectorExpandedListOK(t *testing.T) {
+	validators := createDummyNodesList(10, "shard_0")
+	weights := createDummyWeights(10, 32, 2, nextWeightArithmeticProgression)
+	hasher := &sha256.Sha256{}
+
+	selector, err := NewSelectorExpandedList(validators, weights, hasher)
+	require.Nil(t, err)
+	require.NotNil(t, selector)
+
+	for i := uint32(0); i < uint32(len(validators)); i++ {
+		offset := uint32(0)
+		if i > 0 {
+			offset = weights[i-1]
+		}
+		for j := i + offset; j < weights[i]; j++ {
+			require.Equal(t, selector.expandedList[j], i)
+		}
+	}
+}
+
+func TestSelectorExpandedList_IsInterfaceNil(t *testing.T) {
```

Comment:
you can remove this test and add `require.False(t, check.IfNIl(..)` on the the 'OkValues' test for constructor, and check for true in test for bad constructors

Reply:
I prefer to have separate test per method.